recipe/build.sh

```diff
@@ -1,6 +1,6 @@
 
 if [ "$(uname)" == "Darwin" ]; then
-    ENABLE_FORTRAN=OFF
+    ENABLE_FORTRAN=ON
```

Comment:
I guess this will simply be ignored if there is no fortran compiler around?

Reply:
It should error if there's no fortran compiler around for mac. compiler presence is assured by new lines 21 and 91. The reason it doesn't complain for Windows is the build/skip line for the fortran output and ENABLE_FORTRAN isn't turned on in bld.bat.

Reply:
In build.sh, the aarch and other  branches of the if/else could be merged, I think,  and just have a specialized value for derivative level

Reply:
Windows won't have a libxc package that gets updated (until we someday add a Windows fortran compiler build). But in practice, folks will continue to get the C-lib-only v6.1 package currently published when they request libxc win-64.

Reply:
Sorry for the barrage -- I keep thinking of additional aspects that might address your question. 

To my understanding, the outer level build/host requirements (before the outputs field) are what's present for the conda-build. So that has the merged requirements so that build.sh/bld.bat see C compilers, Fortran compilers, and Python, and the line you highlighted for Mac can be fulfilled. The package builds once with everything (C/F/Py) enabled. Then the files field for each output name divides the spoils into packages. Then the output name build/host/run requirements are processed into the dependencies to be assigned to each output; so libxc-c depends on libgcc, libxc depends on libgcc and libgfortran, and pylibxc depends on libgcc and the build-time python.

Reply:
Thanks @loriab for this clear explanation, indeed I have/had not yet quite grasped how multiple outputs work under the hood.

Just one follow-up question: if it's the outer level build requirements that define the environment at build time, do we still need `build` requirements in the output packages?
I get that you still want `host`/`run` requirements, but I guess the `build` requirements then don't do anything?

Reply:
I read [here](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#subpackage-requirements) that the `build` requirements of the subpackage are available during the "packaging stage" of that subpackage (perhaps some scripts you want to run to postprocess the `files`).

As far as I understand the docs, it seems to me that in our case both the `build` and `host` requirements of the outputs can be dropped

Reply:
P.S. If you need some dependencies for the `test` stage of the outputs, there is also
```
test:
  requires:
    - package
```

Reply:
> I get that you still want host/run requirements on the output packages, but I guess the build requirements then don't do anything (and might, over time, become outdated vs the top-level build requirements)?

It does reason that build/host are unneeded, but this is where `run_exports` is doing its work. If libxc had needed MKL to compile, we'd have put `mkl-devel` in the host. Because that mkl recipe has a run_exports, placing `mkl-devel` in the libxc recipe's host section would have effectively imposed a properly pinned runtime mkl dependence upon libxc. That's the way most run_exports (incl. the build/run_exports ones in this recipe) work. There's another type: "strong" run_exports that jump two sections from build to run. They're responsible for compiler('c') imposing libgcc runtime dependence. 

So as far as I know, of outer build/host/run and inner build/host/run, outer run are the only section that can be skipped.

Reply:
@ltalirz, revised understanding of outer/inner requirements based on the build reduction work below: split output packages can each have their own subpackage build script, or a single build script for all (situation above), or a mix (situation below). If each subpackage has its own build script, no outer requirements block is necessary. In the mixed case, the output requirements covers the subpackages built with the common build script, then the "files" field partitions them. So for the below, the outer requirements/build has c & fortran compilers so that packages libxc-c and libxc (Fortran) can be build in one pass per arch with the common build.sh/bld.bat. The python dep has been removed from the outer and confined to the inner pylibxc requirements/host so that each arch iterates over pythons and runs the trivial build_py.sh/bld_py.sh.